app/controllers/authenticate_controller.rb

```diff
@@ -180,6 +219,50 @@ def k8s_inject_client_cert
 
   private
 
+  def render_authn_token(authn_token)
+    content_type = :json
+    if encoded_response?
+      logger.debug(LogMessages::Authentication::EncodedJWTResponse.new)
+      content_type = :plain
+      authn_token = ::Base64.strict_encode64(authn_token.to_json)
+      response.set_header("Content-Encoding", "base64")
+    end
+    render(content_type => authn_token)
+  end
+
+  def log_audit_success(
+    authn_params:,
+    audit_event_class:
+  )
+    ::Authentication::LogAuditEvent.new.call(
+      authenticator_name: authn_params.authenticator_name,
+      webservice: authn_params.webservice,
+      client_ip: authn_params.client_ip,
+      role: authn_params.role,
+      account: authn_params.account,
+      username: authn_params.username,
+      audit_event_class: audit_event_class,
+      error: nil
+    )
+  end
+
+  def log_audit_failure(
+    authn_params:,
+    audit_event_class:,
+    error:
+  )
+    ::Authentication::LogAuditEvent.new.call(
+      authenticator_name: authn_params.authenticator_name,
+      webservice: authn_params.webservice,
+      client_ip: authn_params.client_ip,
+      role: authn_params.role,
+      account: authn_params.account,
+      username: authn_params.username,
```

Comment:
how about we pass this as `authn_params` (or `authentication_params`) instead of passing the specific fields? we can avoid a [data clump](https://refactoring.guru/smells/data-clumps) by doing this